src/cp/cp_action.c

```diff
@@ -0,0 +1,270 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#ifdef HAVE_CONTROL_POINT
+
+/* cp_action.c: Definitions for control point actions. */
+/* This file must be edited when a new control point action is created. */
+
+/*
+ * The names below are for an action named "Example action".
+ *
+ * Each action has:
+ * - Action config parsing function (Must be manual):
+ * __wt_control_point_config_action_example_action().
+ *
+ * Some actions (only "Wait for trigger" so far) have additional functions. These functions are
+ * named __wt_control_point_*_example_action.
+ */
+
+/*
+ * __wt_verbose levels used in WT_VERB_CONTROL_POINT logging.
+ * -3 ERROR - Internal error in control point framework.
+ * -2 WARNING - Recoverable internal error in control point framework.
+ * -1 NOTICE: Not used.
+ * 0 INFO - Not used.
+ * 1 DEBUG_1 - When triggered.
+ * 2 DEBUG_2 - Done waiting.
+ * 3 DEBUG_3 - When not triggered.
+ * 4 DEBUG_4 - Start waiting, Other details.
+ * 5 DEBUG_5 - When disabled.
+ */
+
+/*
+ * Action: Sleep: Delay at a specific code location during an execution via __wt_sleep.
+ */
+/* Action config parsing function. */
+/*
+ * __wt_control_point_config_action_sleep --
+ *     Configuration parsing for control point action "Sleep: Delay at a specific code location
+ *     during an execution".
+ *
+ * @param session The session. @param data Return the parsed data in here. @param cfg The
+ *     configuration strings.
+ */
+int
+__wt_control_point_config_action_sleep(
+  WT_SESSION_IMPL *session, WT_CONTROL_POINT *data, const char **cfg)
+{
+    /* TODO. Replace these hard wired values with control point action configuration parsing. */
+    /* TODO. When the hard wire is removed, delete this function from func_ok() in dist/s_void. */
+    WT_CONTROL_POINT_ACTION_SLEEP *action_data;
+    WT_UNUSED(session);
+    WT_UNUSED(cfg);
+
+    action_data = (WT_CONTROL_POINT_ACTION_SLEEP *)(data + 1);
```

Comment:
Correct me if I'm wrong, here we have a `__wt_conn_control_point_action_data_sleep` and it's being provide as a `WT_CONTROL_POINT*`? 

Why not just do `action_data = ((WT_CONN_CONTROL_POINT_ACTION_DATA_SLEEP *)data)->action_data;`? The pointer arithmetic is not clear.

Reply:
I agree with Luke's comment that the pointer arithmetic is not obvious here, and could go wrong if the code were  subsequently changed to that "+1" is not the right calculation.

A slight correction to Luke's suggestion, as the address of action_data is needed, is:

`action_data = &((WT_CONN_CONTROL_POINT_ACTION_DATA_SLEEP *)data)->action_data;`

and would, I think, make the code clearer.

Reply:
Yes. That is necessary for WT_CONTROL_POINT_REGISTRY.data to refer to the control point data for all control points. I also did it here.

This is a standard C idiom.  Incrementing a pointer whether done by *ptr++ = content; or (ptr + 1) advances the pointer by the size of the object pointed at.

I decided I don't need to hide the actual parameter type like was needed in WT_CONTROL_POINT_REGISTRY.
Done. Changed it.